### DISCUSSION TITLE
OEA governance

### DISCUSSION THREAD

**Discussion by genegc** [microsoft/OpenEduAnalytics#325] [General]
OEA has shifted to a joint governance model run by a technical board with representatives from Microsoft, Kwantum Analytics, Authentica Solutions, and Edwire.

We are in the process of developing a more formal governance plan - with the intent of following a model similar to https://github.com/fairlearn/governance

For now, we will meet every 2 weeks to review pull requests and discuss any other topics.

**Comment by genegc** [comment]
Notes from meeting on 9/15/2023

Attendance - JF, Viraj, Chad, Jethro, Gene
Summary:
- we agreed to meet every 2 weeks to review PR's
- we'll develop a common understanding of the criteria we're using in reviewing PR's, and then switch to doing PR reviews and merges asynchronously
- Edwire would like to discuss what the release cycle will be, and what the criteria should be for releasing v1.0
- Gene will add JF as an OEA repo contributor
- JF and Viraj will consolidate the PR's submitted by Edwire, to be reviewed on our next call
- Chad will review pr#297 power bi report
- Gene will review pr#245 from Abhinav, #289 ed-fi doc update from Ayuna, #306 license

**Comment by genegc** [comment]
Notes from meeting on 9/29/2023
Attendance - Chad, Gene
Summary:
- Neither Chad nor Gene have the ability to grant access to JF to be an OEA contribute, and Gene doesn't have pr approval permissions. Gene will reach out to Jethro to get permissions fixed.
- Chad and Gene reviewed, commented on, and merged several pr's (see image below)
- Gene will follow up with Abhinav to have him agree to the contributor license in order to get his contribution merged in

![image](https://github.com/microsoft/OpenEduAnalytics/assets/1812048/c6baf932-9565-44ae-98a8-adf55b1ada9e)

**Comment by drchadvidden** [comment]
Notes from meeting on 10/27/2023

Attendance - JF, Viraj, Chad
Summary:

- EdFi module pull request review (#323)
    - Discussed and approve module tech assets (notebooks and pipelines)
    - Some minor OEA framework changes including flexibility in OEA setup resource group naming convention, suggest small update to OEA main readme to ensure OEA setup process documentation is correct
    - JF/Viraj to update OEA main readme to ensure setup docs correct, add to pull request then approve
    - Suggest in future that JF/Viraj update EdFi module readme docs to align with Canvas module so tech assets are easily shared with OEA community
    
- Planning discussion, next meeting topics
    - Focus on making EdFi module more robust  and production ready (add error handling, absorb functionality to OEA framework, make generalized and transfer to Canvas etc)
    - Integrate SAP into OEA framework
    - Integrate databricks into OEA framework

**Reply by drchadvidden** [reply]
2023-10-27
PR 323
Changes Summary Till Date
OEA - OEA_py
Ingest function to deal with json lines
OEA Utils: create_spark_schemas_from_definitions - checking for the presence of the x-Ed-Fi-explode
Ed-Fi - Notebooks
edfi_py
EdFiOEAChild: Child Class of Base OEA for Ed-Fi Module
EdFiClient: Class for Landing and interacting with Ed-Fi API
edfi_fetch_urls
EdFiApiManager: Class to manage Ed-Fi urls (python substitute for Synapse Web Activities)
EdFi_Refine
Schema Changes in s2r (SQL Friendly Data Model)
Data Partitioning (+ directory changes)
Extended Lake DB Tables
Ad Hoc Changes - LastModifiedDate Inclusion, etc.
Ed-Fi - Pipelines
Removed Web Activities for initializing urls (now done in notebooks)
Parameterization Changes
Small Pool as default in all Pipelines
Ed-Fi - setup.sh
Provisioning additional Ed-Fi Assets (edfi_py, edfi_fetch_urls)

**Comment by genegc** [comment]
Notes from mtg on 3/22/2024
in attendance: Jethro, JF, Gene

- There will be changes needed for the Education Insights module
- We are ready to move forward with the changes to the governance structure for OEA
- The initial steering committee will be comprised of: Jethro Seghers (Microsoft), Jean-Francois Guertin (Edwire), Chad Vidden (Kwantum), Gene Garcia (Authentica Solutions)
- We will begin meeting weekly for now, during this transitional period at least
- Our next steps include
-- communication with the community about the upcoming change
-- migrate OEA repo into OEA github org
-- repo refresh and restructuring
